src/main/java/com/example/onlinebookstore/validation/category/impl/CategoryValidatorImpl.java

```diff
@@ -0,0 +1,38 @@
+package com.example.onlinebookstore.validation.category.impl;
+
+import com.example.onlinebookstore.dto.category.CategoryDto;
+import com.example.onlinebookstore.exception.EntityAlreadyPresentException;
+import com.example.onlinebookstore.exception.InvalidCategoryException;
+import com.example.onlinebookstore.repository.category.CategoryRepository;
+import com.example.onlinebookstore.validation.category.CategoryValidator;
+import com.example.onlinebookstore.validation.common.NullFieldValidator;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+
+@Component
+@RequiredArgsConstructor
+public class CategoryValidatorImpl implements CategoryValidator {
```

Comment:
Null checks are performed by annotations now. This means you have only one valuable method that checks existence of category in DB.
I don't see necessity of separate validator in such case. This just can be private method of relevant service

Reply:
I just wanna left it because. It the second way of Realization of It. I don't want to remove it pls

Reply:
agree, its not necessary to create separate validator